.github/workflows/unit-test.yml

```
@@ -97,61 +107,129 @@ jobs:
           ref: ${{ inputs.branch }}
           fetch-depth: ${{ inputs.checkout_fetch_depth }}
 
+      - name: Determine benchmark name
+        if: ${{ inputs.pipeline_mode != 'unit-tests' }}
+        id: benchmark_name
+        run: |
+          job_name="${{ inputs.job_name }}"
+          _benchmark_name=${job_name//[(,)]/""}
+          _benchmark_name=${_benchmark_name//[(" ")]/"-"}
+          echo "benchmark_name=$_benchmark_name" >> $GITHUB_OUTPUT
+
+      - name: Cache reference benchmarks
+        if: ${{ inputs.pipeline_mode != 'unit-tests' }}
+        id: benchmark-cache
+        uses: actions/cache@v3
+        with:
+          path: ${{ github.workspace}}/benchmark_reference
```

`${{ github.workspace }}` is the default top directory anyways, we can omit.

```suggestion
          path: benchmark_reference
```